restx-i18n/src/test/java/restx/i18n/MessagesRouterTest.java

```diff
@@ -20,18 +20,22 @@ public void should_get_labels_json() throws Exception {
         HttpRequest request = server.client().GET("/api/i18n/labels.json");
 
         assertThat(request.code()).isEqualTo(200);
-        assertThat(request.body()).isEqualTo("{\n" +
+        assertThat(replaceNewLine(request.body())).isEqualTo("{\n" +
```

Comment:
I would maybe use `String.format("%n")` for platform specific line separators instead of introducing a new `replaceNewLine()` method which alters the "code under test".

=> I'd rather transform the code into :
```
assertThat(request.body()).isEqualTo(String.format("{%n" +
                "  \"key1\" : \"value1\",%n" +
                "  \"key2\" : \"hello {{test}}\"%n" +
                "}%n"));
```

WDYT ?

Reply:
Yes it is required there is mixed line endings in the input and output. Remember I was not rewriting those tests I was trying to make them pass. It is a bad idea that the code was checking for line ending this way to begin with but the goal was to make them pass with the least amount of changes but also make Linux, Mac and windows proof

Reply:
Sure, I'm in favor of KISS as well (and that was the reason why we copy/pasted the expectation result, including "hardcoded" `\n`, right into the test)

It's only a matter of philosophy here : I'd prefer to change the expectation rather than the code under test to match Windows compatibility (thus calling `String.format()` + using `%d` in expectation rather than calling `replaceNewLine()` around code under test)

Reply:
This one i forgot actually why I did it this way. It took another two hours to be able to build this in eclipse AGAIN!!!. Anyhow so the reason why I made that change is because the line ending of the origin is inconsistent. I tried your suggestion but the actual output from the method and a few others is like this
window.labels = {\r\n
  "key1" : "value1", \n
  "key2" : "hello {{test}}" \n
};\r\n
So even if u change all the \n to %n as you suggested the method internally is hard coding \n somewhere.

Reply:
I guess this is because labels.json is generated through a property relying on [`restx.i18n.labelsJsTemplate`'s `NamedComponent`](https://github.com/restx/restx/blob/master/restx-i18n/src/main/java/restx/i18n/I18nModule.java#L37) which contains hardcoded `\n` inside it.

Not sure we should fix such things in restx itself, to KISS I would simply `String.format()`'s `%n` in areas that have `\r\n` when you execute test, and keep hardcoded `\n` in the tests expectation when you have a `\n` in the code under test.

It would change `MessagesRouterTest.should_get_labels_js()` with :
```
        assertThat(replaceNewLine(request.body())).isEqualTo(String.format(
                "// RESTX Labels - customize this with restx.i18n.labelsJsTemplate named String\n" +
                "window.labels = {%n" +
                "  \"key1\" : \"value1\",\n" +
                "  \"key2\" : \"hello {{test}}\"\n" +
                "};%n"));
```

Reply:
@fcamblor as per your request I changed all replaceNewLine references to using String.format. I added the commented tests back. Can we move on to the real issue now